src/components/YourCompany/OfficeAddresses.vue

```diff
@@ -282,171 +315,164 @@ export default class OfficeAddresses extends Mixins(CommonMixin, EntityFilterMix
+          ) && this.isSame(
+            this.getOfficeAddresses.registeredOffice.mailingAddress,
+            this.getOfficeAddresses.recordsOffice?.mailingAddress
+          )
+        )
+
+        // compare addresses to set the "inherited records mailing" flag
+        this.inheritRecMailingAddress = this.isSame(
+          this.getOfficeAddresses.recordsOffice?.mailingAddress,
+          this.getOfficeAddresses.recordsOffice?.deliveryAddress
+        )
+      }
+    }
+  }
+
+  /**
+   * When "same as (registry) mailing address" checkbox is changed,
+   * sets the Registered Delivery Address to the Registered Mailing Address.
+   */
   private setDeliveryAddressToMailingAddress (): void {
     if (this.inheritMailingAddress) {
       this.deliveryAddress = { ...this.mailingAddress }
     } else {
-      // Clear to default
+      // clear to default
       this.deliveryAddress = { ...this.defaultAddress }
     }
 
     // Records delivery address also needs to be updated if inherited
     if (this.inheritRegisteredAddress) {
       this.recDeliveryAddress = { ...this.deliveryAddress }
     }
-
-    this.emitAddresses()
   }
 
-  /** Sets the Records office addresses to the Registered office addresses. */
+  /**
+   * When "same as registered address" checkbox is changed,
+   * sets the Records office addresses to the Registered office addresses.
+   */
   private setRecordOfficeToRegisteredOffice (): void {
     if (this.inheritRegisteredAddress) {
-      this.recDeliveryAddress = { ...this.deliveryAddress }
       this.recMailingAddress = { ...this.mailingAddress }
+      this.recDeliveryAddress = { ...this.deliveryAddress }
     } else {
       this.inheritRecMailingAddress = this.inheritMailingAddress
-      // Clear to defaults
+      // clear to default
       this.recMailingAddress = { ...this.defaultAddress }
       this.recDeliveryAddress = { ...this.defaultAddress }
     }
-    this.emitAddresses()
   }
 
-  /** Sets the Records Delivery Address to Records Mailing Address. */
+  /**
+   * When "same as (records) mailing address" checkbox is changed,
+   * sets the Records Delivery Address to Records Mailing Address.
+   */
   private setRecordDeliveryAddressToMailingAddress (): void {
     if (this.inheritRecMailingAddress) {
       this.recDeliveryAddress = { ...this.recMailingAddress }
     } else {
-      // Clear to default
+      // clear to default
       this.recDeliveryAddress = { ...this.defaultAddress }
     }
-    this.emitAddresses()
   }
 
-  private updateAddress (addressChangedEntity: string, baseAddress: AddressIF, newAddress: AddressIF): void {
+  /**
+   * Handles update events from address sub-components.
+   */
+  private updateAddress (addressToUpdate: string, baseAddress: AddressIF, newAddress: AddressIF): void {
     Object.assign(baseAddress, newAddress)
-    switch (addressChangedEntity) {
+    switch (addressToUpdate) {
       case 'mailingAddress':
```

Comment:
I know you didn't change anything here, but in hindsight, we should have done an enum for the different address types. 4 types is enough to make it worthy i think 😆

Reply:
I considered it, and decided I had enough work to do atm 😆

As I'm still working on the unit tests, I'll have another look at this. Might just do it this time.

Reply:
Done. Please see second commit.